packages/plugin-ext/src/hosted/browser/hosted-plugin-informer.ts

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright (C) 2018 Red Hat, Inc. and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+import { injectable, inject, postConstruct } from 'inversify';
+import { StatusBar } from '@theia/core/lib/browser/status-bar/status-bar';
+import { StatusBarAlignment, StatusBarEntry } from '@theia/core/lib/browser';
+import { WorkspaceService } from '@theia/workspace/lib/browser';
+import { HostedPluginServer } from '../../common/plugin-protocol';
+import { ConnectionStatusService, ConnectionState } from '@theia/core/lib/browser/connection-status-service';
+import URI from '@theia/core/lib/common/uri';
+import { FileStat } from '@theia/filesystem/lib/common';
+import { FrontendApplicationStateService } from '@theia/core/lib/browser/frontend-application-state';
+
+/**
+ * Informs the user whether Theia is running with hosted plugin.
+ * Adds 'Development Host' status bar element and appends the same prefix to window title.
+ */
+@injectable()
+export class HostedPluginInformer {
```

Comment:
this class should implement FrontendApplicationContribution
and registered through
   bind(FrontendApplicationContribution).to(HostedPluginInformer).inSingletonScope();

Reply:
Done